docs/core-concepts/networking.md

```diff
@@ -0,0 +1,177 @@
+  The first `/24` of addresses within the overlay network subnet (`10.81.1.X/19`) is reserved for these statically allocated Node overlay network addresses.
+
+  These Node overlay network addresses can be used by both other Nodes and any Service Containers within the same Grid.
+  The Node's overlay network address is used for the Grid's infrastructure services, including Kontena's etcd cluster.
+  Any Kontena Service container can connect to each host Node's overlay network address.
+
+* The Docker gateway address (`172.17.0.1` on `docker0`)
+
+  Each Host Node uses the Docker default bridge network to provide Containers with access to the internal network.
+
+  The Docker gateway address is a also used as the DNS resolver for the Containers, provided by Weave DNS.
+
+### Node Infrastructure Services
+Each host Node runs a number of infrastructure services as Docker containers, using the host network namespace.
+
+| Service   | Protocol | Port | Addresses                 | Description
+|-----------|----------|------|---------------------------|-----------------------
+| Weave DNS | TCP+UDP  | 53   | `172.17.0.1` (`docker0`)  | Weave DNS
+| etcd      | TCP      | 2379 | <ul><li>`127.0.0.1` (`lo`)</li><li>`172.17.0.1` (`docker0`)</li><li>`10.81.0.X` (weave)</li></ul> | etcd Clients
+| etcd      | TCP      | 2380 | `10.81.10.X` (weave)       | etcd Peers
+| Weave Net | TCP      | 6783 | `*`                       | Weave Net Control
+| Weave Net | UDP      | 6783 | `*`                       | Weave Net Data (`sleeve`)
+| Weave Net | UDP      | 6784 | `*`                       | Weave Net Data (`fastdp`)
+
+Only the Weave Net service is externally accessible, which is required for forming the Overlay Network mesh between host Nodes.
+
+## Overlay Network
+
+At startup, the Kontena Agent establishes the overlay network mesh between the Grid's Nodes.
+The overlay network mesh is used to bootstrap the Grid infrastructure, as the host Node's overlay network address is used for infrastructure services such as etcd.
+
+The overlay network is established using the network addresses of the peer Nodes within the Grid, which the Agent receives from the Master at startup.
+The overlay network mesh uses the Private network address of a peer node within the same Region, otherwise the Public network address is used.
+Once the overlay network is started, the host Node's overlay network address is configured using [`weave expose`](https://www.weave.works/docs/net/latest/using-weave/host-network-integration/).
+
+The overlay network is powered by Weave Net, using [Weave's encrypted `sleeve` tunnels](https://www.weave.works/docs/net/latest/using-weave/security-untrusted-networks/) to form a flat Layer 2 network spanning all Grid Nodes and connected Containers.
+
+Alternatively, [Weave's Fast Datapath](https://www.weave.works/docs/net/latest/using-weave/fastdp/) can be used for traffic between Nodes within the Grid's Trusted Subnets.
```

Comment:
Do we already have documentation about how to configure trusted subnets? If yes, a link to it would be nice here.